smartobjects/restitution/__init__.py

```
@@ -71,6 +71,30 @@ def rows(self):
         """
         return self._rows
 
+    @property
+    def df(self):
+        """Returns the result set as a Pandas DataFrame if pandas is present. Otherwise return None.
+
+        Example:
+            >>> resultset.df
+                             year  COUNT(*)
+            0 2016-01-01 05:00:00      2155
+            1 2017-01-01 05:00:00      1232
+        """
+        try:
+            import pandas
+            dataframe = pandas.io.json.json_normalize(self._source, 'rows')
+            dataframe.columns = [col['label'] for col in self._columns]
+            for column in self._columns:
+                if column['type'] == 'datetime':
+                    id_col = column['label']
+                    dataframe[id_col] = pandas.to_datetime(dataframe[id_col])
+            return dataframe
+
+        except ImportError:
+            return None
```

I would raise an exception to avoid silent failure if the user doesn't know the feature is optional.

E.g. `raise ImportError('The 'panda' package is required to use this feature')`